### DISCUSSION TITLE
MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR questions

### DISCUSSION THREAD

**Discussion by oseiler2** [meshtastic/firmware#3734] [General]
I had a look into creating a minimised build and tried using '#define MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR'. Not sure if the intention behind the PR #3337 was just disabling the modules, or also excluding the sensor libraries.

If I understand correctly currently the main effect is that it excludes the EnvironmentTelemetryModule and AirQualityTelemetryModule. But the sources under `modules/Telemetry/Sensor` get still pulled in and compiled, as does all the library dependencies, effectively not really reducing the build size much.

The libraries get included in [`[environmental_base]`](https://github.com/meshtastic/firmware/blob/master/platformio.ini#L115), which is then included in the various `/arch/*.ini` files.

1. Excluding `[environmental_base]` leads to compiler errors, and would also need either 
  `#if !MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR` statements in all the Sensor sources, or a `build_src_filter` based exclusion. Some sensors are used for Telemetry and would need `#if HAS_TELEMETRY` which probably means a `build_src_filter` won't really work that well without refactoring them.
2. `AccelerometerThread` uses sensors that have their libs included in `[environmental_base]`. Should the `AccelerometerThread` also be excluded when `MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR` is defined?
3. Currently [`QMC5883LCompass`](https://github.com/meshtastic/firmware/blob/master/platformio.ini#L99) is not included under `[environmental_base]` - should it be moved there?

Happy to create a PR, but would like some feedback first where this should be heading to.

**Comment by thebentern** [comment]
IMO, if we're excluding environmental sensors to minimize the build, I think it's safe to assume that we don't care about accelerometers or other peripherals.

**Comment by oseiler2** [comment]
PR https://github.com/meshtastic/firmware/pull/3776
